dbt/adapters/bigquery/relation_configs/materialized_view.py

```diff
@@ -0,0 +1,165 @@
+            "max_staleness": config_dict.get("max_staleness"),
+            "allow_non_incremental_definition": config_dict.get(
+                "allow_non_incremental_definition"
+            ),
+            "kms_key_name": config_dict.get("kms_key_name"),
+            "friendly_name": config_dict.get("friendly_name"),
+            "description": config_dict.get("description"),
+            "labels": config_dict.get("labels"),
+        }
+
+        materialized_view: "BigQueryMaterializedViewConfig" = super().from_dict(kwargs_dict)  # type: ignore
+        return materialized_view
+
+    @classmethod
+    def parse_model_node(cls, model_node: ModelNode) -> dict:
+        config_dict = {
+            "materialized_view_name": model_node.identifier,
+            "schema_name": model_node.schema,
+            "database_name": model_node.database,
+            "cluster_by": model_node.config.extra.get("cluster_by"),
+            "partition_by": model_node.config.extra.get("partition_by"),
+            "partition_expiration_date": model_node.config.extra.get("partition_expiration_date"),
+            "refresh_interval_minutes": model_node.config.extra.get("refresh_interval_minutes"),
+            "hours_to_expiration": model_node.config.extra.get("hours_to_expiration"),
+            "max_staleness": model_node.config.extra.get("max_staleness"),
+            "allow_non_incremental_definition": model_node.config.extra.get(
+                "allow_non_incremental_definition"
+            ),
+            "kms_key_name": model_node.config.extra.get("kms_key_name"),
+            "friendly_name": model_node.config.extra.get("friendly_name"),
+            "description": model_node.config.extra.get("description"),
+            "labels": model_node.config.extra.get("labels"),
+        }
+
+        autorefresh_value = model_node.config.extra.get("enabled_refresh")
+        if autorefresh_value is not None:
+            if isinstance(autorefresh_value, bool):
+                config_dict["enable_refresh"] = autorefresh_value
+            elif isinstance(autorefresh_value, str):
+                lower_autorefresh_value = autorefresh_value.lower()
+                if lower_autorefresh_value == "true":
+                    config_dict["enable_refresh"] = True
+                elif lower_autorefresh_value == "false":
+                    config_dict["enable_refresh"] = False
+                else:
+                    raise ValueError(
+                        "Invalide enable_refresh representation. Please used excepted value ex.(True, 'true', 'True')"
+                    )
+            else:
+                raise TypeError("Invalid autorefresh value: expecting boolean or str.")
+
+        return config_dict
+
+    @classmethod
+    def parse_relation_results(cls, relation_results: RelationResults) -> dict:
+        materialized_view: agate.Row = cls._get_first_row(
+            relation_results.get("materialized_view")  # type: ignore[arg-type]
```

Comment:
@mikealfare any ides why removing the type ignore here causes a type error on `_get_first_row` due to `argument 1 to "_get_first_row" of "bigqueryreleationconfigbase" has incompatible type "union[row, table, none]"; expected "table"`

Reply:
`cls._get_first_row()` expects an `agate.Table` as an input, but `RelationResults` is defined as `Dict[str, Union[agate.Row, agate.Table]]` and `.get()` returns an optional `None` if "materialized_view" is not a key in `relation_results`. Hence the type of `relation_results.get("materialized_view")` is `Union[agate.Row, agate.Table, None]`. We just know it's an `agate.Table` because we know what's calling it and we define the key. We could also appease the `mypy` overlords by separating this into two steps and verifying the result is an `agate.Table` before piping it into `_get_first_row`, like this:

```python
materialized_view_config = relation_results.get("materialized_view")
if isinstance(materialized_view_config, agate.Table):
    materialized_view = cls._get_first_row(materialized_view_config)
else:
    raise DbtRuntimeError("...")
```